libjwt/jwt.c

```diff
@@ -1063,9 +1118,37 @@ char *jwt_encode_str(jwt_t *jwt)
 	errno = jwt_encode(jwt, &str);
 	if (errno) {
 		if (str)
-			free(str);
+			jfree(str);
 		str = NULL;
 	}
 
 	return str;
 }
+
+int jwt_set_alloc(jwt_alloc_t palloc, jwt_realloc_t prealloc, jwt_free_t pfree)
+{
+	if (!palloc || !prealloc || !pfree)
```

Comment:
Remove `NULL` checks so that we can do selective override, and/or revert to native allocation.

Reply:
Done. I decided that the all-or-nothing check was unnecessarily complicated. I left it as you suggested where the caller can specify any combination of functions they want.